source/Calamari/ArgoCD/Conventions/ApplicationValidator.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.ArgoCD.Domain;
+using Calamari.ArgoCD.Models;
+
+namespace Calamari.ArgoCD.Conventions
+{
+    static class ApplicationValidator
+    {
+        public static ValidationResult Validate(Application application)
+        {
+            return ValidationResult.Merge(
+                                          ValidateSourceNames(application),
+                                          ValidateUnnamedAnnotationsInMultiSourceApplication(application)
+                                         );
+        }
+
+        static ValidationResult ValidateSourceNames(Application application)
+        {
+            var groupedByName = application.Spec.Sources.GroupBy(s => s.Name.ToApplicationSourceName());
```

Comment:
nit: your life is probably easier if you just treat them as string

Reply:
yeah...nah... 😁 

The tiny type extension deals with whitespace for me